### PR TITLE
add raw data validation to the bench ingestion test

### PR DESCRIPTION
**Pull request by imakhlin**
Note, new configuration option added to `tsdb-bench-test-config.yaml`
```
# validate raw data correctness
ValidateData: true
```

### CODE REVIEW

File: test/benchmark/BenchmarkIngest_test.go [hunk 1]
```diff
@@ -122,6 +123,16 @@ func BenchmarkIngest(b *testing.B) {
...
+
+	if testConfig.ValidateRawData {
+		for metricName := range samplesModel {
+			tsdbtest.ValidateRawData(b, adapter, metricName, testStartTimeMs, testEndTimeMs, isValidSequence)
+		}
+	}
+}
+
+func isValidSequence(prev float64, current float64) bool {
+	return current-prev == 1
```
Comment (gtopper) [inline test/benchmark/BenchmarkIngest_test.go]: I guess it works so nice, but float equality can be tricky :)

**Comment by imakhlin** [comment]
Note: running the BenchIngest test shows huge decrease of ingestion performance 

> Test complete. 89856000 samples added to tsdb-BenchmarkIngest-1000000-2018-09-19T19_17_59_03_00
>...
> ok      github.com/v3io/v3io-tsdb/test/benchmark        1595.822s

**Comment by imakhlin** [comment]
Update: after reverting the ingestion algorithm back and reducing `BatchSize` to `32` the performance (ingestion throughput) got much better (140s vs 1595s): 
>About to run 50000000 ingestion cycles from 2018-09-18T00:21:33+03:00 [1537219293603] to 2018-09-20T00:21:33+03:00 [1537392093603]. Max samples count: 34560
> ...
>        BenchmarkIngest_test.go:137: PASS: data consistency test for n=1000000
>        tsdbtest.go:160: PASS: total count of samples, i.e. [89856000(actualCount) !=89856000(expected)]
>        ... [output truncated]
>PASS
>ok      github.com/v3io/v3io-tsdb/test/benchmark        140.646s

File: pkg/tsdb/tsdbtest/tsdbtest.go [hunk 2]
```diff
@@ -145,16 +145,58 @@ func ValidateCountOfSamples(t testing.TB, adapter *V3ioAdapter, metricName strin
...
+			t.Fatal(set.Err(), "failed to get next element from result set")
+		}
+
+		series := set.At()
+		iter := series.Iterator()
+		for iter.Next() {
+			if iter.Err() != nil {
+				t.Fatal(set.Err(), "failed to get next time-value pair from iterator")
+			}
+			_, currentValue := iter.At()
```
Comment (gshatz) [inline pkg/tsdb/tsdbtest/tsdbtest.go]: please validate the timestamps are consecutive

File: pkg/tsdb/tsdbtest/tsdbtest.go [hunk 3]
```diff
@@ -145,16 +145,58 @@ func ValidateCountOfSamples(t testing.TB, adapter *V3ioAdapter, metricName strin
...
 		t.Fatal(set.Err())
 	}
 
-	if expected != actual {
-		t.Fatalf("Check failed: actual result is not as expected [%d(actual) != %d(expected)]", actual, expected)
+	if expected != actualCount {
+		t.Fatalf("Check failed: actualCount result is not as expected [%d(actualCount) != %d(expected)]", actualCount, expected)
+	}
+
+	t.Logf("PASS: total count of samples, i.e. [%d(actualCount) != %d(expected)]", actualCount, expected)
```
[the remark this answers is not in the document]
Comment (imakhlin) [reply pkg/tsdb/tsdbtest/tsdbtest.go]: typo, will fix it

**Pull request merged**